### PR TITLE
Add Player leveling, item storage/consumption, stamina

### PR DESCRIPTION
**Pull request by LukeFZ**

### CODE REVIEW

File: DragaliaAPI/Features/Item/ItemService.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        logger.LogDebug("Processing items {@useItems}", items);
+
+        UseItemEffect effect = UseItemEffect.None;
+        int totalQuantity = 0;
+
+        foreach (AtgenUseItemList item in items)
+        {
+            UseItemData data = MasterAsset.UseItem[item.item_id];
+            if (effect == UseItemEffect.None)
+                effect = data.ItemEffect;
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/Item/ItemService.cs:36]: are we expecting to only get items of the same effect?

File: DragaliaAPI/Features/Player/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            if (!MasterAsset.UserLevel.TryGetValue(data.Level + 1, out UserLevel? next))
+                break;
+
+            if (current.TotalExp + current.NecessaryExp > data.Exp)
+                break;
+
+            data.Level++;
+            totalReward += WyrmiteLevelUpReward;
+
+            await AddStamina(StaminaType.Single, next.StaminaSingle);
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/Player/UserService.cs:208]: does this need to add wings too?

File: DragaliaAPI/Features/Item/ItemService.cs (hunk 1 above)
Comment (LukeFZ) [reply DragaliaAPI/Features/Item/ItemService.cs:36]: item/use_recovery_stamina doesn't return an array of AtgenRecoverData so the game prevents you from mixing them

File: DragaliaAPI/Features/Dungeon/DungeonRecordController.cs [hunk 3]
```diff
@@ -72,6 +74,33 @@ PlayRecord playRecord
 
         bool isFirstClear = oldQuestData is null || oldQuestData?.PlayCount == 0;
 
+        if (
+            !isFirstClear && (playRecord.is_clear == 1 || session.QuestData.IsPayForceStaminaSingle)
```
[the remark this answers is not in the document]
Comment (LukeFZ) [reply DragaliaAPI/Features/Dungeon/DungeonRecordController.cs:78]: Have not verified this but I thought the wave quests did, the ones where stamina is also consumed even if you fail

File: DragaliaAPI/Features/Player/UserService.cs (hunk 2 above)
Comment (LukeFZ) [reply DragaliaAPI/Features/Player/UserService.cs:208]: yes

File: DragaliaAPI/Features/Dungeon/DungeonRecordController.cs [hunk 4]
```diff
@@ -72,6 +74,33 @@ PlayRecord playRecord
...
+            StaminaType type = StaminaType.None;
+            int amount = 0;
+
+            if (isMulti)
+            {
+                if (session.IsHost)
+                {
+                    type = StaminaType.Multi;
+                    amount = session.QuestData.PayStaminaMulti;
+                }
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/Dungeon/DungeonRecordController.cs:92]: Other players also have getherwings deducted

File: DragaliaAPI/Features/Dungeon/DungeonRecordController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (SapiensAnatis) [reply DragaliaAPI/Features/Dungeon/DungeonRecordController.cs:78]: Looks like is_clear is set as 1 for incomplete facility clears, I suggest we remove it from this condition if we don't know what it does

**Review by SapiensAnatis — APPROVED** [review]

**Pull request merged**